File: CAGradientLayer Example ObjC/CAGradientLayer Example ObjC/KHGradientView.m
//
//  KHGradientView.m
//
//  Created by Kyle Howells on 09/06/2020.
//  Copyright © 2020 Kyle Howells. All rights reserved.
//

#import "KHGradientView.h"

@implementation KHGradientView
@dynamic layer;

+(Class)layerClass{
    return [CAGradientLayer class];
}

//-(CAGradientLayer*)gradientLayer{
//	return (CAGradientLayer*)[self layer];
//}
@end

File: CAGradientLayer Example Swift/CAGradientLayer Example Swift/KHGradientView.swift
//
//  KHGradientView.swift
//
//  Created by Kyle Howells on 15/06/2020.
//  Copyright © 2020 Kyle Howells. All rights reserved.
//

import UIKit

class KHGradientView: UIView {
	
	override open class var layerClass: AnyClass {
	   return CAGradientLayer.classForCoder()
	}
	
	var gradientLayer : CAGradientLayer {
		return self.layer as! CAGradientLayer
	}
	
}
